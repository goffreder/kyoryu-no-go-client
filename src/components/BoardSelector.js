import React, { Component } from 'react';
import { number, func } from 'prop-types';

export default class extends Component {
    static propTypes = {
        size: number.isRequired,
        initBoard: func.isRequired,
    };

    handleClick = () => {
        this.props.initBoard(this.props.size);
    };

    render() {
        return (
            <div className="col s12 m4">
                <div className="card blue-grey darken-1">
                    <div className="card-content" />
                    <div className="card-action center-align">
                        <button
                            className="waves-effect waves-light btn blue-grey btn-large darken-4"
                            onClick={this.handleClick}
                        >{`Init ${this.props.size}x${
                            this.props.size
                        } board`}</button>
                    </div>
                </div>
            </div>
        );
    }
}
