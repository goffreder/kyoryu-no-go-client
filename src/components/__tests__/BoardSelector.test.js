import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-testing-library';

import BoardSelector from '../BoardSelector';

describe('BoardSelector component', () => {
    it('displays a board selector', () => {
        expect(
            renderer
                .create(<BoardSelector size={9} initBoard={() => {}} />)
                .toJSON(),
        ).toMatchSnapshot();
    });

    it('calls "initBoard" prop on button click', () => {
        const initBoard = jest.fn();
        const { getByText } = render(
            <BoardSelector size={9} initBoard={initBoard} />,
        );

        fireEvent.click(getByText('Init 9x9 board'));
        expect(initBoard).toHaveBeenCalledTimes(1);
    });
});
