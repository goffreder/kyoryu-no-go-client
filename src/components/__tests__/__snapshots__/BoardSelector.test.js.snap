// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardSelector component displays a board selector 1`] = `
<div
  className="col s12 m4"
>
  <div
    className="card blue-grey darken-1"
  >
    <div
      className="card-content"
    />
    <div
      className="card-action center-align"
    >
      <button
        className="waves-effect waves-light btn blue-grey btn-large darken-4"
        onClick={[Function]}
      >
        Init 9x9 board
      </button>
    </div>
  </div>
</div>
`;
